.github/workflows/ci.yml

```diff
@@ -269,6 +269,22 @@ jobs:
         with:
           token: ${{ secrets.GITHUB_TOKEN }}
 
+  wasm-integration:
+    name: "wasm-integration"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: rui314/setup-mold@v1
+      - uses: actions/checkout@v4
+      - uses: ./.github/actions/setup-rust
+        with:
+          targets: "wasm32-wasip1"
+      - name: Setup Wasmer
+        uses: wasmerio/setup-wasmer@v3.1
+      - run: cargo build --target wasm32-wasip1
```

Comment:
I tried many times to get this to work properly with wasm32-unknown-unknown but seemingly it builds a module with wasmbindgen imports that I can't seem to shim in and is targeted at running in the browser. I didn't have much luck taking the resulting bundle and running with with node/bun either. Perhaps there's a better way but compiling as a WASI binary was the only thing I was able to get working in CI